src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -122,19 +253,23 @@ private static WellKnownType GetTupleType(int arity)
                                                             WellKnownType.System_ValueTuple_T1_T2_T3_T4,
                                                             WellKnownType.System_ValueTuple_T1_T2_T3_T4_T5,
                                                             WellKnownType.System_ValueTuple_T1_T2_T3_T4_T5_T6,
-                                                            WellKnownType.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7 };
+                                                            WellKnownType.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7,
+                                                            WellKnownType.System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest };
 
         /// <summary>
         /// Find the constructor for a well-known ValueTuple type of a given arity.
+        ///
         /// For example, for arity=2:
         /// returns WellKnownMember.System_ValueTuple_T1_T2__ctor
+        ///
+        /// For arity=12:
+        /// return System_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__ctor
         /// </summary>
         internal static WellKnownMember GetTupleCtor(int arity)
         {
             if (arity > 7)
```

Comment:
It doesn't feel appropriate to have the fallback since we now have full support for "long" tuples. I don't think an assert   would address this concern.